### ISSUE TITLE
Unable to preview Encrypted PDF after upgrading to 3.0.0-beat.1

### ISSUE THREAD

**Issue by pdfdroid** [barteksc/AndroidPdfViewer#451]
Hi,
I have implemented the PDF password dialog, with help of error msg returned in 2.7.x version.
Since I required some of your updates in 3.0.0-beat.1, i upgraded.
But this encrypted pdf preview is now coming as a white screen.

Thank you for your understandings.

**Comment by barteksc** [comment]
Is there any log in logcat?

**Comment by 1stmetro** [comment]
Beta 2, passwords working for me, what do you mean by preview ?

**Comment by pdfdroid** [comment]
@barteksc 

_Please find the logs when opening an encrypted pdf as below:_

D/jniPdfium: Init FPDF library
D/jniPdfium: Destroy FPDF library
E/PDFView: load pdf error
	com.shockwave.pdfium.PdfPasswordException: Password required or incorrect password.
		at com.shockwave.pdfium.PdfiumCore.nativeOpenDocument(Native Method)
		at com.shockwave.pdfium.PdfiumCore.newDocument(PdfiumCore.java:133)
		at com.github.barteksc.pdfviewer.source.FileSource.createDocument(FileSource.java:38)
		at com.github.barteksc.pdfviewer.DecodingAsyncTask.doInBackground(DecodingAsyncTask.java:49)
		at com.github.barteksc.pdfviewer.DecodingAsyncTask.doInBackground(DecodingAsyncTask.java:25)
		at android.os.AsyncTask$2.call(AsyncTask.java:333)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
		at java.lang.Thread.run(Thread.java:764)

**Comment by pdfdroid** [comment]
@1stmetro 
3.00-beta2 is released?
I am using 3.00-beta.1 rlease.
Preview is nothing but opening an Encrypted PDF file.(which is exported as a PDF from MSWord )

**Comment by jaideepreddy** [comment]
After updating to 3.0.0.-beta.1, Facing the same issue for Encrypted files

**Comment by 1stmetro** [comment]
I would post your code as it clearly says password issues, I have no problems with passwords.

**Comment by pdfdroid** [comment]
@1stmetro 
Can you please share, the approach, how you are identifying the encrypted file?
Because, when i am trying with this code 
```.onError(new OnErrorListener() {
@override
public void onError(Throwable t) {
if (t.getMessage().contains("Password required or incorrect password")) {
},
```
I am failed to open the encrypted pdf files(white screen is displayed)

**Comment by barteksc** [comment]
@pdfdroid 
Indeed, `OnErrorListener` is not called. It will be fixed in next release.
But please, do not identify exception by its message...

**Comment by pdfdroid** [comment]
@barteksc 
Thank you for your response.
When i can expect the next release?
##But please, do not identify exception by its message...
Please Share the other posible inputs if any to identify encrypted pdf's...

**Comment by 1stmetro** [comment]
Simply set the password, and catch if its wrong using the exception not the string.
.password(password)

 if (t.toString().contains("PdfPasswordException")) {

**Comment by 1stmetro** [comment]
Maybe this may help a little more....

.onError(new OnErrorListener() {
                                        @Override
                                        public void onError(Throwable t) {
                                            if (t.toString().contains("PdfPasswordException")) {
You now found the password issue, what would you like to do with it ?
}

I'm sure this will help resolve your issue.

**Comment by barteksc** [comment]
This is Java, you may compare types by `instanceof` or by `Class#isAssignableFrom(Class)`.
`if (t instanceof PdfPasswordException)` or `PdfPasswordException.class.isAssignableFrom(t.getClass())`

**Comment by 1stmetro** [comment]
or you could do it that way ! thanks @barteksc

**Comment by 1stmetro** [comment]
Ahh, i see the problem now the password code is no longer working correctly it does'nt catch the error when the password is wrong as there cant be any return showing an error just a white screen, even using if (t instanceof PdfPasswordException) the code is slightly broken from what it once was, but works fine with the correct password.

**Comment by barteksc** [comment]
Fixed in 3.0.0-beta.3
